### PR TITLE
Dir grouping/v1

### PR DESCRIPTION
**Pull request by shivan1b**
Closes redmine ticket: https://redmine.openinfosecfoundation.org/issues/3069

### CODE REVIEW

**Comment by shivan1b** [comment]
Hey @catenacyber !
Could you please have a look at this?

File: run.py [hunk 1]
```diff
@@ -803,8 +803,11 @@ def main():
         if dirpath == tdir:
             continue
 
-        # We only want to go one level deep.
-        dirnames[0:] = []
+        # Check if there are sub-test directories
+        if "test.yaml" in filenames or "check.sh" in filenames:
+            dirnames[0:] = []
```
Comment (catenacyber) [inline run.py]: To me, this is python dark magic (`dirnames` value seems not used) but it was already like this

File: run.py [hunk 2]
```diff
@@ -764,6 +764,8 @@ def main():
                         help="Exit on test failure")
     parser.add_argument("--testdir", action="store",
                         help="Runs tests from custom directory")
+    parser.add_argument("--skip-tests", nargs="?", default=None,
+                        help="Runs tests from custom directory")
```
Comment (catenacyber) [inline run.py]: The long description is copy-pasted 
It should be updated to `Skips tests with this pattern`

File: run.py (hunk 1 above)
Comment (catenacyber) [reply run.py]: Maybe we can add a comment that `dirnames` value reference gets used by `os.walk`

File: run.py (hunk 2 above)
Comment (shivan1b) [reply run.py]: Oops. Fixing this. Thanks.

File: run.py (hunk 1 above)
Comment (shivan1b) [reply run.py]: OK. Thanks.

**Comment by shivan1b** [comment]
Repalced with #115

**Pull request closed without merging**